### PR TITLE
20483 entity auth update to work with alternate_name

### PR DESCRIPTION
**Pull request by vysakh-menon-aot**
*Issue #:* /bcgov/entity#20483

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: queue_services/entity-auth/src/entity_auth/resources/worker.py [hunk 1]
```diff
@@ -232,22 +243,45 @@ def create_affiliation(legal_entity: LegalEntity, filing: Filing):
...
 
-    if len(entity_roles) == 0:
-        return pass_code
+    if business.is_alternate_name_entity and business.entity_type == BusinessCommon.EntityTypes.SOLE_PROP.value:
+        if business.is_owned_by_legal_entity_person:
+            pass_code = business.legal_entity.last_name + ", " + business.legal_entity.first_name
+            if business.legal_entity.middle_initial:
+                pass_code = pass_code + " " + business.legal_entity.middle_initial
+        elif business.is_owned_by_legal_entity_org or business.is_owned_by_colin_entity:
+            pass_code = business.legal_name
```
Comment (argush3) [inline queue_services/entity-auth/src/entity_auth/resources/worker.py:257]: for the `business.is_owned_by_colin_entity` scenario, you can get the organization name via `business.colin_entity.organization_name`

File: queue_services/entity-auth/src/entity_auth/resources/worker.py [hunk 2]
```diff
@@ -232,22 +243,45 @@ def create_affiliation(legal_entity: LegalEntity, filing: Filing):
...
 
-    party = entity_roles[0].related_entity
+    elif entity_roles := EntityRole.get_entity_roles(business.id, end_date):
+        party = entity_roles[0].related_entity
```
Comment (argush3) [inline queue_services/entity-auth/src/entity_auth/resources/worker.py:260]: for the scenario where the partner is in COLIN, entity_roles[0].related_colin_entity will be populated.  And the entry will linked to `colin_entities` table.

File: queue_services/entity-auth/src/entity_auth/resources/worker.py (hunk 1 above)
Comment (vysakh-menon-aot) [reply queue_services/entity-auth/src/entity_auth/resources/worker.py:257]: Are you suggesting this?
```suggestion
        elif business.is_owned_by_legal_entity_org:
            pass_code = business.legal_name
        elif business.is_owned_by_colin_entity:
            pass_code = business.colin_entity.organization_name
```

Comment (argush3) [reply queue_services/entity-auth/src/entity_auth/resources/worker.py:257]: yes

**Review by argush3 — APPROVED** [review]

**Pull request merged**